### PR TITLE
Implemented User object

### PR DESCRIPTION
**Pull request by sattmaunders**
UserTransformer to build User object from JSON
Implemented User to hold data instead
Conversion takes place in UserHelper.java

### CODE REVIEW

File: mobile/src/main/java/com/standapp/activity/MainActivity.java [hunk 1]
```diff
@@ -107,6 +108,7 @@
 //    private OnDataPointListener mListener;
     private boolean connectedToFitAPI = false;
     private JSONObject user;
```
Comment (johnsintal) [inline mobile/src/main/java/com/standapp/activity/MainActivity.java]: We can get rid of this, right?

File: mobile/src/main/java/com/standapp/backend/UserHelper.java [hunk 2]
```diff
@@ -27,7 +29,10 @@ public void checkIfUserIsCreated(final String userEmail, final UserHelperListene
 
         Response.Listener<JSONObject> successListener = new Response.Listener<JSONObject>() {
             public void onResponse(JSONObject response) {
-                userHelperListener.onUserExists(response);
+                //Convert response to User object
+                UserTransformer userTransformer = new UserTransformer();
```
Comment (johnsintal) [inline mobile/src/main/java/com/standapp/backend/UserHelper.java]: This UserTransformer should be injected at the activity scope level. The class UserHelper has a depedency on UserTransformer, so you want to inject UserTransformer into UserHelper  via constructor. Look at the @provides annotation.

Comment (johnsintal) [inline mobile/src/main/java/com/standapp/backend/UserHelper.java]: We're gonna use the constructor injection, not @Inject for this, look at Backendserver or UserInfo, how they get injected.

File: mobile/src/main/java/com/standapp/activity/MainActivity.java (hunk 1 above)
Comment (sattmaunders) [inline mobile/src/main/java/com/standapp/activity/MainActivity.java]: Definitely

**Pull request merged**